src/types/model.ts

```diff
@@ -985,6 +994,15 @@ export interface LucidModel {
     options?: ModelAdapterOptions
   ): Promise<null | InstanceType<T>>
 
+  /**
+   * Find one using a clause or fail
+   */
+  findByOrFail<T extends LucidModel>(
+    this: T,
+    clause: Record<string, unknown>,
```

Comment:
Hi, @RomainLanz!

Any reasons why `clause` is not type-safe? Looks like it is possible to infer at least keys.